.circleci/scripts/git-diff-develop.js

```diff
@@ -0,0 +1,76 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+const fs = require('fs');
+const path = require('path');
+
+
+async function fetchWithDepthIncrement(depth) {
+  try {
+    await exec(`git fetch --depth ${depth} origin develop`);
+    await exec(`git fetch --depth ${depth} origin ${process.env.CIRCLE_BRANCH}`);
+    return true;
+  } catch (error) {
+    console.error(`Failed to fetch with depth ${depth}:`, error.message);
+    return false;
+  }
+}
+
+async function gitDiffWithRetry() {
+  const depths = [1, 10, 100];
+  let diffOutput = '';
+
+  for (const depth of depths) {
+    try {
+      console.log(`Attempting git diff with depth ${depth}...`);
+      await fetchWithDepthIncrement(depth);
+
+      const { stdout: diffResult } = await exec(`git diff --name-only origin/develop...${process.env.CIRCLE_BRANCH}`);
+        diffOutput = diffResult;
+        break;
+      } catch (error) {
+        if (error.message.includes('no merge base')) {
+          console.error(`Error 'no merge base' encountered with depth ${depth}. Incrementing depth...`);
+        } else {
+          throw error;
+        }
+      }
+  }
+
+  if (!diffOutput) {
+    // If no merge base found, fetch the entire history
+    await exec(`git fetch --unshallow origin develop`);
+
+    const { stdout: diffResult } = await exec(`git diff --name-only origin/develop...${process.env.CIRCLE_BRANCH}`);
+    diffOutput = diffResult;
+
```

Comment:
this new update has been tested and works correctly - check the logs [here](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/87981/workflows/4095b4ea-f19d-498b-8aed-7a694670ba68/jobs/3230260)




![Screenshot from 2024-06-18 18-49-23](https://github.com/MetaMask/metamask-extension/assets/54408225/7bff2d87-97f8-4d96-bc00-b83f8c94c4c0)